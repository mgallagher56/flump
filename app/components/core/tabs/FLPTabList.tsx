import React from 'react';

import type { TabListProps } from '@chakra-ui/react';
import { TabList } from '@chakra-ui/react';

export const FLPTabList: React.FC<TabListProps> = (props) => {
  return <TabList {...props}>{props.children}</TabList>;
};
