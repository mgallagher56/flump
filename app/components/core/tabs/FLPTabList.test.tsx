import React from 'react';

import { Tabs } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import FLPTabList from './FLPTabList';

const tabWrapper = (children: React.ReactNode): React.ReactElement => <Tabs>{children}</Tabs>;

describe('FLPTabList', () => {
  test('renders correctly', () => {
    const { container } = render(
      tabWrapper(
        <FLPTabList>
          <div>Tab 1</div>
          <div>Tab 2</div>
          <div>Tab 3</div>
        </FLPTabList>
      )
    );
    expect(container).toMatchSnapshot();
  });
});
