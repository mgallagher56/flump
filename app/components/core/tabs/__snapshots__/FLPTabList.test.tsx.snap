// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPTabList > renders correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-79elbk"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-k008qs"
      role="tablist"
    >
      <div>
        Tab 1
      </div>
      <div>
        Tab 2
      </div>
      <div>
        Tab 3
      </div>
    </div>
  </div>
</div>
`;
